gems/aws-sdk-core/spec/aws/credential_resolution_chain_spec.rb

```diff
@@ -93,6 +93,36 @@ module Aws
         ).to eq('AR_AKID')
       end
 
+      it 'prefers sso credentials over assume role' do
+        expect(SSOCredentials).to receive(:new).with(
+          sso_start_url: 'START_URL',
+          sso_region: 'us-east-1',
+          sso_account_id: 'SSO_ACCOUNT_ID',
+          sso_role_name: 'SSO_ROLE_NAME'
+        ).and_return(
+          double(
+            'creds',
+            set?: true,
```

Comment:
wouldn't `set?` return true already because you are mocking credentials?

Reply:
No - the set? method gets called on this double and if its not mocked it will fail.